unicodetools/src/main/java/org/unicode/text/UCD/MakeUnicodeFiles.java

```diff
@@ -799,6 +808,391 @@ private static void generateArabicShaping(String filename) throws IOException {
+                this(
+                        header,
+                        new UnicodeSet(
+                                unicodeSet,
+                                new ParsePosition(0),
+                                VersionedSymbolTable.forDevelopment()),
+                        dispreferredStringPredicate);
+            }
+
+            DoNotEmitSubsubsection(
+                    String header,
+                    UnicodeSet unicodeSet,
+                    Predicate<String> dispreferredStringPredicate) {
+                if (dispreferredStringPredicate == null) {
+                    this.dispreferredStrings = unicodeSet;
+                } else {
+                    this.dispreferredStrings = new UnicodeSet();
+                    for (final String dispreferred : unicodeSet) {
+                        if (dispreferredStringPredicate.test(dispreferred)) {
+                            this.dispreferredStrings.add(dispreferred);
+                        }
+                    }
+                }
+                this.header = header;
+            }
+
+            void printHeader() {
+                pw.println("# " + header.replace("\n", "\n# "));
+            }
+
+            UnicodeSet dispreferredStrings;
+            String header;
+        }
+        class DoNotEmitSubsection extends DoNotEmitSubsubsection {
+            DoNotEmitSubsection(String header, String unicodeSet) {
+                super(header, unicodeSet, null);
+            }
+
+            DoNotEmitSubsection(
+                    String header,
+                    String unicodeSet,
+                    Predicate<String> dispreferredStringPredicate) {
+                super(header, unicodeSet, dispreferredStringPredicate);
+            }
+
+            DoNotEmitSubsection(
+                    String header,
+                    UnicodeSet unicodeSet,
+                    Predicate<String> dispreferredStringPredicate) {
+                super(header, unicodeSet, dispreferredStringPredicate);
+            }
+
+            @Override
+            void printHeader() {
+                pw.println();
+                super.printHeader();
+            }
+        }
+        class DoNotEmitSection {
+            DoNotEmitSection(
+                    String header,
+                    Comparator<String> compareDispreferredSequences,
+                    DoNotEmitSubsubsection... subsections) {
+                this.subsections = subsections;
+                this.compareDispreferredSequences = compareDispreferredSequences;
+                this.header = header;
+            }
+
+            DoNotEmitSubsubsection[] subsections;
+            Comparator<String> compareDispreferredSequences;
+            String header;
+        }
+        // TODO(egg): These sections seem like an unmaintainable mess (complete with numbered
+        // references to core specification tables).  Can we come up with something a little more
+        // systematic?
```

Comment:
Ad hoc file structure is fine while a file is maintained manually. Parsing and regenerating moves the goal posts.

I thought that @roozbehp had some kind of tooling for DoNotEmit.txt though?

Anyway, I suggest you create a PAG ticket to discuss options.